File: priv/combined_metrics/samples/variable_naming/name_is_too_long/good/session.rb
# Session and auth management with concise, clear variable names.
# GOOD: current_user, max_retries, selected_product — short but descriptive.

class SessionGood
  def start_session(email, password)
    user = fetch_user_by_email(email)
    is_valid = verify_password(password, user[:password_hash])

    raise 'Invalid credentials' unless is_valid

    token = generate_session_token
    expires_at = Time.now + 86_400

    { token: token, expires_at: expires_at, user_id: user[:id] }
  end

  def validate_session(token)
    session = lookup_session(token)
    now = Time.now

    raise 'Session expired' if now > session[:expires_at]

    session
  end

  def refresh_session(old_token)
    current_session = validate_session(old_token)
    new_token = generate_session_token
    expires_at = Time.now + 86_400

    { token: new_token, expires_at: expires_at, user_id: current_session[:user_id] }
  end

  def list_active_sessions(user_id)
    max_retries = 3
    sessions = fetch_all_sessions(user_id, max_retries)
    now = Time.now

    sessions.select { |session| session[:expires_at] > now }
  end

  def current_user(token)
    session = validate_session(token)
    fetch_user_by_id(session[:user_id])
  end

  def invalidate_all_sessions(user_id)
    active_sessions = list_active_sessions(user_id)
    active_sessions.each { |session| delete_session(session[:id]) }
  end

  private

  def fetch_user_by_email(email) = { id: 1, email: email, password_hash: 'hash' }
  def fetch_user_by_id(id) = { id: id }
  def verify_password(password, _hash) = password.length > 0
  def generate_session_token = SecureRandom.hex(32)
  def lookup_session(_token) = { user_id: 1, expires_at: Time.now + 3600 }
  def fetch_all_sessions(_user_id, _max_retries) = []
  def delete_session(_session_id) = nil
end

File: priv/combined_metrics/samples/variable_naming/name_is_too_long/good/session.js
// Session and auth management with concise, clear variable names.
// GOOD: currentUser, maxRetries, selectedProduct — short but descriptive.

async function startSession(email, password) {
  const user = await fetchUserByEmail(email);
  const isValid = await verifyPassword(password, user.passwordHash);

  if (!isValid) {
    throw new Error('Invalid credentials');
  }

  const token = generateSessionToken();
  const expiresAt = new Date(Date.now() + 86_400_000);

  return { token, expiresAt, userId: user.id };
}

async function validateSession(token) {
  const session = await lookupSession(token);
  const now = new Date();

  if (now > session.expiresAt) {
    throw new Error('Session expired');
  }

  return session;
}

async function refreshSession(oldToken) {
  const currentSession = await validateSession(oldToken);
  const newToken = generateSessionToken();
  const expiresAt = new Date(Date.now() + 86_400_000);

  return { token: newToken, expiresAt, userId: currentSession.userId };
}

async function listActiveSessions(userId) {
  const maxRetries = 3;
  const sessions = await fetchAllSessions(userId, maxRetries);
  const now = new Date();

  return sessions.filter(session => session.expiresAt > now);
}

async function currentUser(token) {
  const session = await validateSession(token);
  return fetchUserById(session.userId);
}

async function invalidateAllSessions(userId) {
  const activeSessions = await listActiveSessions(userId);
  await Promise.all(activeSessions.map(session => deleteSession(session.id)));
}

async function fetchUserByEmail(email) { return { id: 1, email, passwordHash: 'hash' }; }
async function verifyPassword(password) { return password.length > 0; }
function generateSessionToken() { return Math.random().toString(36).slice(2); }
async function lookupSession() { return { userId: 1, expiresAt: new Date(Date.now() + 3600_000) }; }
async function fetchAllSessions() { return []; }
async function fetchUserById(id) { return { id }; }
async function deleteSession() {}

module.exports = { startSession, validateSession, refreshSession, listActiveSessions, currentUser, invalidateAllSessions };
